cf-app/src/main/kotlin/space/crowdforce/controllers/ProjectController.kt

```diff
@@ -3,23 +3,215 @@ package space.crowdforce.controllers
 import io.swagger.annotations.Api
 import io.swagger.annotations.ApiOperation
 import org.springframework.http.MediaType
-import org.springframework.web.bind.annotation.GetMapping
-import org.springframework.web.bind.annotation.RequestMapping
-import org.springframework.web.bind.annotation.RestController
-import space.crowdforce.controllers.model.ActivityUI
-import space.crowdforce.controllers.model.ProjectUI
-import java.time.LocalDateTime.now
+import org.springframework.web.bind.annotation.*
+import space.crowdforce.controllers.model.*
+import space.crowdforce.domain.Activity
+import space.crowdforce.domain.Project
+import space.crowdforce.domain.User
+import space.crowdforce.domain.geo.Location
+import space.crowdforce.service.activity.ActivityService
+import space.crowdforce.service.project.ProjectService
+import space.crowdforce.service.user.UserService
+import java.lang.RuntimeException
+import java.security.Principal
 
 @Api(value = "/api/v1/projects", description = "")
 @RestController
 @RequestMapping("/api/v1/projects")
-class ProjectController {
+class ProjectController(
+    private val projectService: ProjectService,
+    private val activityService: ActivityService,
+    private val userService: UserService
+) {
     @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
     @ApiOperation(value = "")
-    suspend fun getProjects() = listOf(
-        ProjectUI(1, "test 1", 59.984945f, 30.343947f, false, emptyList()),
-        ProjectUI(2, "test 2", 59.983000f, 30.350230f, true, listOf(
-            ActivityUI(1232, "Activity 1", 59.981611f, 30.356375f, now(), now().plusDays(5), true)
-        ))
-    )
+    fun getProjects(principal: Principal?): List<ProjectUI> {
+        val userId = if (principal != null) userService.getUserIdByName(principal.name) else null
+
+        return projectService.getAllProjectAggregation(userId).map { map(it) }
+    }
+
+    @PostMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
```

Comment:
You can move `produces` and `consumes` to class level under RequestMapping

Reply:
fixed